homeassistant/components/tautulli/sensor.py

```diff
@@ -56,84 +70,41 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Create the Tautulli sensor."""
-
-    name = config[CONF_NAME]
-    host = config[CONF_HOST]
-    port = config[CONF_PORT]
-    path = config[CONF_PATH]
-    api_key = config[CONF_API_KEY]
-    monitored_conditions = config.get(CONF_MONITORED_CONDITIONS, [])
-    users = config.get(CONF_MONITORED_USERS, [])
-    use_ssl = config[CONF_SSL]
-    verify_ssl = config[CONF_VERIFY_SSL]
-
-    session = async_get_clientsession(hass=hass, verify_ssl=verify_ssl)
-
-    api_client = PyTautulli(
-        api_token=api_key,
-        hostname=host,
-        session=session,
-        verify_ssl=verify_ssl,
-        port=port,
-        ssl=use_ssl,
-        base_api_path=path,
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
     )
 
-    coordinator = TautulliDataUpdateCoordinator(hass=hass, api_client=api_client)
-
-    async_add_entities(
-        new_entities=[
-            TautulliSensor(
-                coordinator=coordinator,
-                name=name,
-                monitored_conditions=monitored_conditions,
-                usernames=users,
-            )
-        ],
-        update_before_add=True,
-    )
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tautulli sensor."""
+    sensors = [
+        TautulliSensor(
+            hass.data[DOMAIN][entry.entry_id],
+            description,
+            entry.entry_id,
+        )
+        for description in SENSOR_TYPES
+    ]
+
+    async_add_entities(sensors, True)
```

Comment:
`async_add_entities` takes a generator, so do the loop in the first argument instead of storing it like this.